maven-surefire-plugin/src/main/java/org/apache/maven/plugin/surefire/SurefirePlugin.java

```diff
@@ -253,6 +253,12 @@
     @Parameter( property = "surefire.rerunFailingTestsCount", defaultValue = "0" )
     private int rerunFailingTestsCount;
 
+    /**
+     * The number of flakes after which the overall test suite will be considered a failure.
+     */
+    @Parameter( property = "surefire.failOnFlakeCount", defaultValue = "0" )
```

Comment:
Could we specify here whether `0` means a single retry will fail the build (i.e. similar to disabling retries completely, I guess), or whether this disables the feature (which seems to be the case based on https://github.com/apache/maven-surefire/pull/333/files#diff-bf7f174589af8f08638da17d6aefbe897133b7b77ff363a1e12439c37ea8683eR146, which seems weird to me, maybe we'd have to come up with a better name)